source/elements/oneDAL/source/algorithms/clustering/kmeans.rst

```diff
@@ -1,153 +1,494 @@
 
 ---
 API
 ---
 
-Descriptor
-----------
-.. code-block:: c++
-
-   namespace onedal::kmeans {
+Methods
+-------
+::
 
    namespace method {
-   struct lloyd {};
-   using by_default = lloyd;
+      struct lloyd {};
+      using by_default = lloyd;
    } // namespace method
 
+.. namespace:: onedal::kmeans::method
+.. struct:: lloyd
+
+   Tag-type that denotes `Lloyd's method`_.
+
+
+.. type:: by_default = lloyd
+
+   Alias tag-type for the Lloyd's method.
+
+
+Descriptor
+----------
+::
+
    template <typename Float = float,
              typename Method = method::by_default>
    class desc {
    public:
-      double get_gamma() const;
+      desc();
+
+      int64_t get_cluster_count() const;
+      int64_t get_max_iteration_count() const;
       double get_accuracy_threshold() const;
-      std::int64_t get_cluster_count() const;
-      std::int64_t get_max_iteration_count() const;
 
-      desc& set_gamma(double);
+      desc& set_cluster_count(int64_t);
+      desc& set_max_iteration_count(int64_t);
       desc& set_accuracy_threshold(double);
-      desc& set_cluster_count(std::int64_t);
-      desc& set_max_iteration_count(std::int64_t);
    };
 
-   } // namespace onedal::kmeans
+.. namespace:: onedal::kmeans
+.. class:: template<typename Float = float, \
+                    typename Method = method::by_default> \
+           desc
+
+   :tparam Float: The floating-point type that the algorithm uses for
+                  intermediate computations. Can be :expr:`float` or :expr:`double`.
+
+   :tparam Method: Tag-type that specifies variant of K-Means algorithm. Can be
+                   :expr:`method::lloyd` or :expr:`method::by_default`.
+
+   .. function:: desc()
+
+      Creates new instance of descriptor with the default attribute values.
+
+   .. member:: std::int64_t cluster_count = 2
+
+      The number of clusters :math:`k`.
+
+      Getter & Setter
+         | ``std::int64_t get_cluster_count() const``
+         | ``desc& set_cluster_count(std::int64_t)``
+
+      Invariants
+         | :expr:`cluster_count > 0`
+
+
+   .. member:: std::int64_t max_iteration_count = 100
+
+      The maximum number of iterations :math:`T`.
+
+      Getter & Setter
+         | ``std::int64_t get_max_iteration_count() const``
+         | ``desc& set_max_iteration_count(std::int64_t)``
+
+      Invariants
+         | :expr:`max_iteration_count >= 0`
+
+
+   .. member:: double accuracy_threshold = 0.0
+
+      The threshold :math:`\varepsilon` for the stop condition.
+
+      Getter & Setter
+         | ``double get_accuracy_threshold() const``
+         | ``desc& set_accuracy_threshold(double)``
+
+      Invariants
+         | :expr:`accuracy_threshold >= 0.0`
 
 
 Model
 -----
-.. code-block:: c++
+::
 
    class model {
    public:
+      model();
+
       const table& get_centroids() const;
+      int64_t get_cluster_count() const;
    };
 
+.. class:: model
+
+   .. function:: model()
+
+      Creates model with default attribute values.
```

Comment:
what is 'attribute' in this context?

Reply:
Attribute aggregates setter and setter functions to a single entity called _attribute_, we'll need to introduce this terminology somewhere in common section